README.md

```diff
@@ -67,12 +67,15 @@ form](https://coder.com/contact) or send an email to support@coder.com.
 | imagePullPolicy | string | Sets the policy for pulling a container image across all services. | `"Always"` |
 | ingress.additionalAnnotations | list | Deprecated. Please use `ingress.annotations`. | `[]` |
 | ingress.annotations | object | Additional annotations to be used when creating the ingress. These only apply to the Ingress Kubernetes kind. The annotations can be used to specify certificate issuers or other cloud provider-specific integrations. | `{}` |
+| ingress.deployment | object | Options used to configure the Deployment of the default ingress controller. | `{"kind":"Deployment"}` |
+| ingress.deployment.kind | string | How to deploy the built-in ingress. Can be "Deployment" or "DaemonSet". | `"Deployment"` |
```

Comment:
Not sure I understand this - in the case where we want a cluster-wide ingress controller, doesn't it make more sense for users to deploy their own and set `ingress.enable` and `ingress.useDefault` to false?

Reply:
I think this would be optimal, but doing so currently is not that easy. Right now we expect customers to use our ingress unless it's not possible to do so. At a certain point we start reimplementing the `ingress-nginx` helm chart, but I don't see a good alternative atm

Reply:
Once we have the agent architecture & coderd work that @f0ssel and @kylecarbs are driving, it seems like we won't need path-based routing anymore, so then we can conceivably drop the ingress too, right? That would allow us to remove this stuff from our chart

Reply:
Yea p2p could let us drop nginx-ingress for more cloud native solutions, but it would still be a breaking change. We would still need to deploy an Ingress, but the controller backing it can be much more flexable.